examples/ui-demo/src/components/small-cards/SolanaNftCard.tsx

```diff
@@ -101,12 +99,13 @@ export const SolanaNftCard = () => {
 
       const mint = stakeAccount.publicKey;
       const tx = await sendTransactionAsync({
-        preSend: async (transaction, next) => {
-          if ("version" in transaction) {
-            transaction.sign([stakeAccount]);
-          } else {
-          }
-          return next(transaction, PK);
+        transactionComponents: {
+          preSend: async (transaction) => {
+            if ("version" in transaction) {
+              transaction.sign([stakeAccount]);
+            }
+            return signers.fromSigner(transaction);
+          },
```

Comment:
this is something that app devs need to know about?

is this not something that our hook can do to simplify things?

Reply:
This is in the case we want to over write the sending, for example on this one we have one more signer, and then we need the logic for the previous. It will fail on the simulation for the send transaction if we forgot the original from signer

And in the case of poke the duck, there is no signer except the sponsor, so the overwrite is just an identity function.

Reply:
So, the components of the sendSolanaTransaction
(transfer? ) -> instructions --transactionBuilder-> Transaction --preSend-> Broadcast
And now as params are the components that make up this function in the cases we want custom logic, it is easy for a one off change now.

Reply:
```mermaid
stateDiagram-v2
[*] --> instructions
[*] --> transfer
transfer --> instructions
instructions --> Transaction : transactionBuilder
Transaction --> broadcastedResult : preSend
```